crud/common/utils.lua

```diff
@@ -199,18 +199,78 @@ function utils.tarantool_supports_uuids()
     return enabled_tarantool_features.uuids
 end
 
-function utils.convert_operations(user_operations, space_format)
+local function exists_operations_id_map(operations)
+    local map = {}
+    for _, operation in ipairs(operations) do
+        map[operation[2]] = true
+    end
+
+    return map
+end
+
+local function add_nullable_fields_rec(operations, operations_id_map, space_format, tuple, id)
+    if id < 2 or tuple[id - 1] ~= box.NULL then
+        return operations
+    end
+
+    if space_format[id - 1].is_nullable and not operations_id_map[id - 1] then
+        table.insert(operations, {'=', id - 1, box.NULL})
+        return add_nullable_fields_rec(operations, operations_id_map, space_format, tuple, id - 1)
+    end
+
+    return operations
+end
+
+-- Tarantool < 2.3 has no fields `box.error.NO_SUCH_FIELD_NO` and `box.error.NO_SUCH_FIELD_NAME`.
+if _TARANTOOL >= "2.3" then
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD_NO or err_code == box.error.NO_SUCH_FIELD_NAME
+    end
+else
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD
+    end
+end
+
+function utils.add_intermediate_nullable_fields(operations, space_format, tuple)
+    if tuple == nil then
+        return operations
+    end
```

Comment:
Is this check really needed?

As I understand if you try to update tuple that doesn't actually exist you don't have any error at all:
```
tarantool> box.space.tasks:update({10000000000}, {{'=', 'abc.d.e.f.g.h', 1000}})
---
...

tarantool> box.space.tasks:update({10000000000}, {{'+', 'abc.d.e.f.g.h', 1000}})
---
...
```

Reply:
I need this check because in ``add_nullable_fields_recursive`` function I am referring to the tuple by index